refactor chaincode operator

AurelienGasser opened the pull request:
## Changes

This PR moves the following calls from the appchannel operator to the chaincode operator, which removes a bit of  code/complexity.

```
peer lifecycle chaincode checkcommitreadiness
peer lifecycle chaincode querycommitted
peer chaincode invoke --isInit
```

## Side note about explicit v.s. implicit options (Edit: Outdated, see comments below)

 I didn't have to use discovery (as [suggested here](https://github.com/SubstraFoundation/hlf-k8s/pull/102#issuecomment-734149787)) to obtain the peer address and the TLS root cert:

```
--peerAddresses ${PEER_HOST}:${PEER_PORT} 
--tlsRootCertFiles tlsRootCert-$org.crt
```
The reason why we don't need to specify these values is that they are the "default" values taken from `core.yaml` and `/var/hyperledger/tls/client/pair`.

Question : Is it cleaner to pass paramters explicitly (via `--peerAddresses` et `--tlsRootCertFiles`), or to use "default" values as described above?

In any event **the other `peer` calls in the chaincode operator (listed below) also don't specify `--peerAddresses` and `--tlsRootCertFile`**. That's not new from this PR.

```
peer lifecycle chaincode install
peer lifecycle chaincode approveformyorg
etc...
```

You can see [in the doc](https://hyperledger-fabric.readthedocs.io/en/release-2.2/commands/peerlifecycle.html) that these commands also support `--peerAddresses` and `--tlsRootCertFile` arguments (but our code uses defaults instead).

We might choose to pass these options explicitly everywhere, or to stick with using defaults as in this PR.

This can be addressed either as part of this PR, or separately. Please comment accordingly.

ClementGautier reviewed (dismissed):
Looks great !

ClementGautier commented:
Regarding your question on the arguments I'd say: use env variables when possible to keep the code easy to read and maintain. Basically use CORE_PEER_ADDRESS / CORE_PEER_TLS_ROOTCERT_FILE etc.
![image](https://user-images.githubusercontent.com/660091/100847110-a8a7eb00-347f-11eb-9c3d-f9c0f099f42a.png)

Kelvin-M reviewed (changes requested):
Nice work, but your `Side note about explicit v.s. implicit options` is not really true ^^

I think you read my [suggestion](https://github.com/SubstraFoundation/hlf-k8s/pull/102#issuecomment-734149787) too quickly, or maybe I don't explain the idea properly. You're confusing `core.yaml` default values to connect to **a** Peer with `--peerAddresses` `--tlsRootCertFiles` to endorse **a** Peer transaction by multiple  external Peer**s**

`--peerAddresses` `--tlsRootCertFiles` is not just two arguments encapsulated in $ENDORSEMENT, it's repeated (`--peerAddresses` `--tlsRootCertFiles`) along all Peers connection like for a Peer1 which try to endorse chaincode commit:

`ENDORSEMENT="--peerAddresses Peer2:7054 --tlsRootCertFiles tlsRootCert-Peer2.crt --peerAddresses Peer3:7054 --tlsRootCertFiles tlsRootCert-Peer3.crt --peerAddresses Peer4:7054 --tlsRootCertFiles tlsRootCert-Peer4.crt"`

You may have not seen any issue because for multiple examples (2 orgs no ca, 2 orgs, 4 orgs), and by default in `configmap-fabric.yaml`, chaincode lifecycle policy is `ANY`
```
      Policies: &ApplicationDefaultPolicies
        LifecycleEndorsement:
            Type: ImplicitMeta
            Rule: "ANY Endorsement"
```

But, for 3 or 4 Orgs majority example, your code won't work, as transaction will be only endorsed by one Peer (itself) , chaincode lifecycle policy is `MAJORITY`
```
appPolicies: |-
    LifecycleEndorsement:
        Type: ImplicitMeta
        Rule: "MAJORITY Endorsement"
```

Kelvin-M commented on charts/hlf-k8s/templates/deployment-chaincode-operator.yaml:
```diff
@@ -167,6 +170,105 @@ spec:
+                  done < <(echo "$DISCOVERY_PEER" | jq -r '.[].MSPID')
+
+                  peer lifecycle chaincode commit \
+                      --signature-policy "OR('MyOrg1MSP.member','MyOrg2MSP.member','MyOrg3MSP.member','MyOrg4MSP.member')" \
```
Hardcoded policy ?

Kelvin-M commented on charts/hlf-k8s/templates/deployment-chaincode-operator.yaml:
```diff
@@ -167,6 +170,105 @@ spec:
+                      --cafile /var/hyperledger/tls/ord/cert/cacert.pem \
+                      --keyfile /var/hyperledger/tls/client/pair/tls.key \
+                      --certfile /var/hyperledger/tls/client/pair/tls.crt \
+                      -o network-orderer-hlf-ord.orderer:7050 \
```
Hardcoded orderer ?

Kelvin-M commented on charts/hlf-k8s/templates/deployment-chaincode-operator.yaml:
```diff
@@ -167,6 +170,105 @@ spec:
+
+                  peer lifecycle chaincode commit \
+                      --signature-policy "OR('MyOrg1MSP.member','MyOrg2MSP.member','MyOrg3MSP.member','MyOrg4MSP.member')" \
+                      --channelID mychannel \
```
Channel and chaincode properties are hardcoded !

AurelienGasser commented:
This is ready for another review

e2e tests: :heavy_check_mark: 

@Kelvin-M as discussed, I extracted the `sequence` changes and moved them to a [new PR](https://github.com/SubstraFoundation/hlf-k8s/pull/107) (I'll let you take it up from there)

Kelvin-M commented:
Rebased
